### PR TITLE
Issue 120

### PR DESCRIPTION
**Pull request by cmoulliard**
- Add 2 new variables/properties
```
quarkus.kubernetes.env.vars.GITHUB_REPO=${github.repo}
quarkus.kubernetes.env.vars.GIT_SHA_COMMIT=${git.sha.commit}
```
- Inject the properties part of the HomeResource class to inject them to the qute template
- Include the URL link on the home page to get the gtihub commit/version

### CODE REVIEW

**Review by aureamunoz — COMMENTED** [review]
I'm OK with the chages but the CI failed, could you take a look?

File: scripts/primaza.sh [hunk 1]
```diff
@@ -49,7 +43,9 @@ pe "mvn clean install -DskipTests -Ppush-images,kubernetes -Dquarkus.container-i
    -Dquarkus.container-image.group=${REGISTRY_GROUP} \
    -Dquarkus.container-image.tag=${IMAGE_VERSION} \
    -Dquarkus.container-image.insecure=true \
-   -Dquarkus.kubernetes.ingress.host=${INGRESS_HOST}"
+   -Dquarkus.kubernetes.ingress.host=${INGRESS_HOST} \
+   -Dgit.sha.commit=$(git rev-parse --short HEAD) \
+   -Dgithub.repo=https://github.com/halkyonio/primaza-poc"
```
Comment (cmoulliard) [inline scripts/primaza.sh:48]: Is it correct s job fails ? @Sgitario

**Review by aureamunoz — COMMENTED** [review]
LGTM but CI failed, could you please take a look?

File: scripts/primaza.sh (hunk 1 above)
Comment (cmoulliard) [reply scripts/primaza.sh:48]: How can you provide default values then ?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -69,7 +69,9 @@ jobs:
             -Dquarkus.container-image.insecure=true \
             -Dquarkus.helm.repository.push=true \
             -Dquarkus.helm.repository.url=http://localhost:8080/api/charts \
-            -Dquarkus.helm.repository.type=CHARTMUSEUM
+            -Dquarkus.helm.repository.type=CHARTMUSEUM \
+            -Dgithub.repo=${{ github.repository }} \
+            -Dgit.sha.commit=$(git rev-parse --short HEAD)
```
Comment (Sgitario) [inline .github/workflows/pr.yml:74]: in the pull request events, the latest sha commit is in `github.event.pull_request.head.sha` (more info in https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)

File: servicebox-app/src/main/java/io/halkyon/resource/page/HomeResource.java [hunk 3]
```diff
@@ -2,14 +2,23 @@
...
 import io.quarkus.qute.TemplateInstance;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 
 @Path("/home")
 public class HomeResource {
+
+    @ConfigProperty(name = "github.repo")
```
Comment (Sgitario) [inline servicebox-app/src/main/java/io/halkyon/resource/page/HomeResource.java:13]: Provide default value:
```java
@ConfigProperty(name = "github.repo", defaultValue = "https://github.com/halkyonio/primaza-poc")
```

File: servicebox-app/src/main/java/io/halkyon/resource/page/HomeResource.java [hunk 4]
```diff
@@ -2,14 +2,23 @@
...
+    String githubRepo;
+    @ConfigProperty(name = "git.sha.commit")
```
Comment (Sgitario) [inline servicebox-app/src/main/java/io/halkyon/resource/page/HomeResource.java:15]: Provide default value:
```java
@ConfigProperty(name = "git.sha.commit", defaultValue = "<< unknown >>")
```

**Review by Sgitario — CHANGES_REQUESTED** [review]

File: scripts/primaza.sh (hunk 1 above)
Comment (Sgitario) [reply scripts/primaza.sh:48]: https://github.com/halkyonio/primaza-poc/pull/121#discussion_r1026196504
and
https://github.com/halkyonio/primaza-poc/pull/121#discussion_r1026196866

Comment (cmoulliard) [reply scripts/primaza.sh:48]: Thanks. This is what I did ;-)

File: .github/workflows/pr.yml (hunk 2 above)
Comment (cmoulliard) [reply .github/workflows/pr.yml]: Fixed: 70e8fd5

**Comment by cmoulliard** [comment]
Help is needed as I dont see how we can fix such an error
```
2022-11-18 09:21:45,005 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, hibernate-orm-rest-data-panache, jdbc-h2, jdbc-postgresql, kubernetes-client, narayana-jta, qute, resteasy, resteasy-jackson, resteasy-multipart, resteasy-qute, scheduler, smallrye-context-propagation, smallrye-health, vertx, webjars-locator]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 11.487 s - in io.halkyon.ClaimingServiceJobTest
[INFO] Running io.halkyon.ClaimsEndpointTest
2022-11-18 09:21:48,388 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /claims failed, error id: 810aa265-f025-46c3-9bec-08b716a50ab4-1: org.jboss.resteasy.spi.UnhandledException: io.quarkus.qute.TemplateException: Rendering error in template [base.html] line 19: Entry "github-repo" not found in the data map in expression {github-repo}
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:368)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:614)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:524)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:564)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:14[78](https://github.com/halkyonio/primaza-poc/actions/runs/3495675500/jobs/5852707344#step:4:79))
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:[83](https://github.com/halkyonio/primaza-poc/actions/runs/3495675500/jobs/5852707344#step:4:84)3)
Caused by: io.quarkus.qute.TemplateException: Rendering error in template [base.html] line 19: Entry "github-repo" not found in the data map in expression {github-repo}
```

@Sgitario

**Comment by Sgitario** [comment]
> Help is needed as I dont see how we can fix such an error
> 
> ```
> 2022-11-18 09:21:45,005 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, hibernate-orm-rest-data-panache, jdbc-h2, jdbc-postgresql, kubernetes-client, narayana-jta, qute, resteasy, resteasy-jackson, resteasy-multipart, resteasy-qute, scheduler, smallrye-context-propagation, smallrye-health, vertx, webjars-locator]
> [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 11.487 s - in io.halkyon.ClaimingServiceJobTest
> [INFO] Running io.halkyon.ClaimsEndpointTest
> 2022-11-18 09:21:48,388 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /claims failed, error id: 810aa265-f025-46c3-9bec-08b716a50ab4-1: org.jboss.resteasy.spi.UnhandledException: io.quarkus.qute.TemplateException: Rendering error in template [base.html] line 19: Entry "github-repo" not found in the data map in expression {github-repo}
> 	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:368)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:614)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:524)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
> 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
> 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
> 	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
> 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
> 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
> 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:564)
> 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
> 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:14[78](https://github.com/halkyonio/primaza-poc/actions/runs/3495675500/jobs/5852707344#step:4:79))
> 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
> 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.base/java.lang.Thread.run(Thread.java:[83](https://github.com/halkyonio/primaza-poc/actions/runs/3495675500/jobs/5852707344#step:4:84)3)
> Caused by: io.quarkus.qute.TemplateException: Rendering error in template [base.html] line 19: Entry "github-repo" not found in the data map in expression {github-repo}
> ```
> 
> @Sgitario

You only added the new properties in the HomeController (to be precise, in the endpoint  `/home`): https://github.com/halkyonio/primaza-poc/pull/121/files#diff-cac6ccb7c438e278cad5bf52cda632aa4bab12c1b68c91aff3c884398ec54979R22
But the `base.html` is rendered in all the endpoints (`/services`, `/clusters`, ...), so you need to add the same data in all the controllers. 
I would suggest to have a service bean with these properties and use it, instead of declaring https://github.com/halkyonio/primaza-poc/pull/121/files#diff-cac6ccb7c438e278cad5bf52cda632aa4bab12c1b68c91aff3c884398ec54979R16 in all the resources.

**Comment by cmoulliard** [comment]
> I would suggest to have a service bean with these properties and use it

Example ? @Sgitario

**Comment by Sgitario** [comment]
> > I would suggest to have a service bean with these properties and use it
> 
> Example ? @Sgitario

```java
@ApplicationScoped
public class RepositoryService {
    @ConfigProperty(name = "github.repo", defaultValue = "http://github.com/halkyonio/primaza-poc")
    String githubRepo;
    @ConfigProperty(name = "git.sha.commit", defaultValue = "6666")
    String gitShaCommit;
}
```

And then, in all the resources:

```java
public class HomeResource {
  @Inject
  RepositoryService repository;
  
   @GET
    public TemplateInstance home() {
        return Templates.App.home().data("title","Home page");
        return Templates.App.home()
                .data("title","Home page")
                .data("repository", repository);
    }
}
```

And finally, in the `base.html`:

```html
 <div class="text-black">Project version : <a href="{repository.githubRepo}/commit/{repository.gitShaCommit}">{repository.gitShaCommit}</a></div>
```

**Comment by cmoulliard** [comment]
Can you review it again as tests are passing ? @aureamunoz @Sgitario

File: .github/workflows/pr.yml [hunk 5]
```diff
@@ -26,7 +26,7 @@ jobs:
           cache: maven
 
       - name: build with maven
-        run: mvn clean install
+        run: mvn clean install -Dgithub.repo=${{ github.repository }} -Dgit.sha.commit=${{ github.event.pull_request.head.sha }}
```
Comment (Sgitario) [inline .github/workflows/pr.yml:29]: I don't think we need to pass these properties here any longer as we're now using defaults.
Also, this ensures the `mvn clean install` works as it is.

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -28,7 +28,7 @@ jobs:
           cache: maven
 
       - name: build with maven
-        run: mvn clean install
+        run: mvn clean install -Dgithub.repo=${{ github.repository }} -Dgit.sha.commit=${{ github.event.pull_request.head.sha }}
```
Comment (Sgitario) [inline .github/workflows/push.yml:31]: I don't think we need to pass these properties here any longer as we're now using defaults.
Also, this ensures the `mvn clean install` works as it is.

**Review by Sgitario — CHANGES_REQUESTED** [review]

File: .github/workflows/pr.yml (hunk 5 above)
Comment (cmoulliard) [reply .github/workflows/pr.yml:29]: That fails without passing such properties as quete reports `io.quarkus.qute.TemplateException: Rendering error in template [base.html] line 19: Property "github.repo" not found on the base object "null" in expression {config:github.repo}`

Comment (cmoulliard) [reply .github/workflows/pr.yml:29]: According to Martin Kouba `{config:github.repo}` does not read the default value from ApplicationProperties but it attempts to read the config value directly :-(

Comment (cmoulliard) [reply .github/workflows/pr.yml]: Should be fixed now: ccee6b0

File: servicebox-app/src/test/java/io/halkyon/qute/ApplicationConfigProperties.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+@QuarkusTest
+public class ApplicationConfigProperties {
+    @Inject
+    ApplicationProperties applicationProperties;
+
+    @Test
+    public void testInjectionAndDefaultValues() {
+        assertEquals("http://github.com/halkyonio/primaza-poc", applicationProperties.getGitHubRepo());
+        assertEquals("666", applicationProperties.getGitShaCommit());
```
Comment (Sgitario) [inline servicebox-app/src/test/java/io/halkyon/qute/ApplicationConfigProperties.java:19]: I like the number

**Review by Sgitario — APPROVED** [review]

**Review by aureamunoz — APPROVED** [review]

**Pull request merged**